langroid/agent/task.py

```diff
@@ -732,8 +733,37 @@ def run(
         if return_type is None:
             return_type = self.default_return_type
 
+        # Take a final strict decoding step
         if return_type is not None and return_type != ChatDocument:
-            return self.agent.from_ChatDocument(final_result, return_type)
+            parsed_result = self.agent.from_ChatDocument(final_result, return_type)
```

Comment:
we need this in the `run_async()` as well -- maybe factor it out as a `_<method>`  and use it in both